ods_ci/tests/Tests/400__ods_dashboard/415__ods_dashboard_projects/415__ods_dashboard_projects_permissions_mgmt.robot

```diff
@@ -76,13 +82,33 @@ Verify User Can Assign Access Permissions To User Groups
     Sleep   5s
     ${USER_C} Should Not Have Access To ${PRJ_USER_B_TITLE}
 
+Verify Project Sharing Does Not Override Dashboard Permissions
+    [Tags]                  Tier1                   Sanity                  ODS-2223
+    # [Setup]                 Run Keywords            Restore Permissions Of The Project
+    # ...                     AND                     Set RHODS Users Group To rhods-users
+    [Setup]                 Set RHODS Users Group To rhods-users
+    Launch Data Science Project Main Page    username=${OCP_ADMIN_USER.USERNAME}    password=${OCP_ADMIN_USER.PASSWORD}
+    ...    ocp_user_auth_type=${OCP_ADMIN_USER.AUTH_TYPE}
+    Assign Admin Permissions To User ${USER_B} in Project ${PRJ_USER_B_TITLE}
+    Assign Edit Permissions To User ${USER_C} in Project ${PRJ_USER_C_TITLE}
+    Remove User From Group    username=${USER_B}    group_name=rhods-users
+    Remove User From Group    username=${USER_B}    group_name=rhods-admins
+    Remove User From Group    username=${USER_C}    group_name=rhods-users
+    Remove User From Group    username=${USER_C}    group_name=rhods-admins
+    Verify No Permissions For User ${USER_B}
+    Verify No Permissions For User ${USER_C}
+    [Teardown]              Run Keywords            Set RHODS Admins Group To system:authenticated
+    ...                     AND                     Set User Groups For Testing
+
 
 *** Keywords ***
 Project Permissions Mgmt Suite Setup    # robocop: disable
     [Documentation]    Suite setup steps for testing DS Projects.
     ...                It creates some test variables and runs RHOSi setup
     Set Library Search Order    SeleniumLibrary
     RHOSi Setup
+    Set Standard RHODS Groups Variables
+    Set Default Access Groups Settings
```

Comment:
this one should in the teardown

Reply:
I had to add it to the Setup too, since we need to make sure the required variables and Users are set correctly before starting the tests. It is safer this way, why do you there an advantage not calling it ?

Reply:
yes but to me it makes sense to restore the default access group in the teardown as well.

Reply:
You right - I run 'Set Default Access Groups Settings' in the new TEST TEARDOWN.
This is to prevent other tests that runs afterwards - but before SUITE TEARDOWN - to fail due to missing default permissions. I will add it to SUITE TEARDOWN as well.